### ISSUE TITLE
Trace events can have negative CPU duration

### ISSUE THREAD

**Issue by dstockwell** [catapult-project/catapult#1506]
Example trace: https://drive.google.com/a/google.com/file/d/0B2AjrrfsiX8QV1lDUjlWSVZVazA/view?usp=sharing

In the browser process there's a `RenderProcessHostImpl::Send` with:

```
Start	
142.042 ms
Wall Duration	
5,836.136 ms
CPU Duration	
-877,330.108 ms
```

I tested for cpuSelfDuration < 0 over 541 android traces. 537 returned true. The above is the largest negative time I've found.

These are the top offenders by count:

* devtools.timeline,v8 :: FunctionCall
* cc,disabled-by-default-devtools.timeline :: AnalyzeTask
* v8 :: v8.newInstance
* cc :: RasterizerTaskImpl::RunOnWorkerThread
* toplevel :: WorkerThread::ThreadMain::Run
* cc,disabled-by-default-devtools.timeline :: RasterTask
* toplevel :: MessageLoop::RunTask
* v8 :: v8.run

**Comment by fmeawad** [comment]
Investigating, in that case, it seems to have 2 slices overlapping, maybe the subtraction was done twice.
![screenshot from 2015-09-23 10 27 08](https://cloud.githubusercontent.com/assets/1760798/10052840/c3e1a944-61dd-11e5-8899-ebf84c6430c5.png)

**Comment by dj2** [comment]
How do we end up with 2 overlapping slices? Is it in the JSON that way or did we do something wrong on import?

It should not be possible to overlap slices ......

**Comment by fmeawad** [comment]
That is what I am looking at

**Comment by fmeawad** [comment]
These 2 slices overlap, 
<table>
<tr><th>Start</th><th>Duration</th></tr>
<tr><td>142.042 ms</td><td>5,836.136 ms</td></tr>
<tr><td>142.164 ms</td><td>8,731.940 ms</td></tr>
</table>

From the JSON
{"pid":5753,"tid":5753,"ts":48339362139,"ph":"X","cat":"renderer_host","name":"RenderProcessHostImpl::Send","args":{},"dur":5836136,"tdur":-877330108,"tts":878762123},
..... 
{"pid":5753,"tid":5753,"ts":48339362261,"ph":"X","cat":"renderer_host","name":"RenderProcessHostImpl::Send","args":{},"dur":8731940,"tdur":-877428475,"tts":878762245},

**Comment by dj2** [comment]
That's crazy. Did we somehow get the wrong dur value along with the wrong tdur value? How could we execute the second Send when the first hasn't finished yet.

**Comment by fmeawad** [comment]
@dstockwell Are these android traces? I need to investigate if we have a random number generator at our hand.

**Comment by dj2** [comment]
From the metadata it looks like it is:

```
"os-arch":"armv7l",
"os-name":"Android",
"product-version":"Chrome/47.0.2512.0"
```

**Comment by dstockwell** [comment]
Yes, I've only looked for this in android traces. Have not checked elsewhere.
